frontend/assets/javascripts/src/modules/campaignCode.es6

```diff
@@ -0,0 +1,36 @@
+/**
+ * Used for storing campaign codes in a cookie for readout server-side.
+ * Necessary for recording campaigns in the soulmates dashboard.
+ *
+ */
+
+import { setCookie } from 'src/utils/cookie';
+import URLSearchParams from 'URLSearchParams';
+
+// Checks the querystring for INTCMP (internal campaign) and records in cookie.
+function recordCampaign () {
+
+	let campaignCode;
+
+	// Don't want JS to break if there is a problem reading query params.
+	// Just don't set the cookie if this happens.
+	try {
+
+		let urlParams = new URLSearchParams(window.location.search);
```

Comment:
Just a small thing, but should you be using `location.search.slice(1)` here? The `slice(1)` is apparently to get rid of the `?`... though it seems to work without the `slice()` in my browser!

http://stackoverflow.com/a/12151322/438886
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams